src/net/benmann/evald/ExpressionParser.java

```diff
@@ -10,12 +12,16 @@
 class ExpressionParser {
     private ExpressionString expression;
     final Evald evald;
+    final Set<Integer> usedIndices = new HashSet<>();
 
     ExpressionParser(Evald evald, ExpressionString expression) {
         this.expression = expression;
         this.evald = evald;
     }
 
+    void addUsedIndex(int index) {
+        usedIndices.add(index);
+    }
 
     Node parse(String expression) {
```

Comment:
Hey Ben, this will miss any usedIndices from the parser it creates.

This fixes it:
```
    Node parse(String expression) {
        ExpressionParser parser = new ExpressionParser(evald, new ExpressionString(expression));

        Node node = parser.parse();
        usedIndices.addAll(parser.usedIndices);

        return node;
    }
```